MueLu doesn't compile....

<!---
Provide a general summary of the issue in the Title above.  If this issue
pertains to a particular package in Trilinos, it's worthwhile to start the
title with "PackageName:  ".
-->
MueLu RefMaxwell doesn't compile 

<!---
Note that anything between these delimiters is a comment that will not appear
in the issue description once created.  Click on the Preview tab to see what
everything will look like when you submit.
-->

<!---
Feel free to delete anything from this template that is not applicable to the
issue you are submitting.
-->

<!---
Replace <teamName> below with the appropriate Trilinos package/team name.
-->
@trilinos/muelu 

<!---
Assignees:  If you know anyone who should likely tackle this issue, select them
from the Assignees drop-down on the right.
-->
@cgcgcg 
<!---
Lables:  Choose any applicable package names from the Labels drop-down on the
right.  Additionally, choose a label to indicate the type of issue, for
instance, bug, build, documentation, enhancement, etc.
-->

## Expectations
<!---
Tell us what you think should happen, how you think things should work, what
you would like to see in the documentation, etc.
-->

## Current Behavior
<!---
Tell us how the current behavior fails to meet your expectations in some way.
-->

```
[0/97] Building CXX object packages/muelu/src/CMakeFiles/muelu.dir/Utils/ExplicitInstantiation/MueLu_CoalesceDropFactory_kokkos.cpp.o
p
In file included from packages/muelu/src/MueLu_Utilities_kokkos.hpp:1:0,
                 from /home/mbetten/Trilinos/Trilinos/packages/muelu/src/Graph/MueLu_CoalesceDropFactory_kokkos_def.hpp:63,
                 from packages/muelu/src/Utils/ExplicitInstantiation/MueLu_CoalesceDropFactory_kokkos.cpp:55:
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Utils/MueLu_Utilities_kokkos_decl.hpp:276:63: error: ‘RealValuedMultiVector’ was not declared in this scope
     static RCP<MultiVector> RealValuedToScalarMultiVector(RCP<RealValuedMultiVector> X);
                                                               ^~~~~~~~~~~~~~~~~~~~~
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Utils/MueLu_Utilities_kokkos_decl.hpp:276:84: error: template argument 1 is invalid
     static RCP<MultiVector> RealValuedToScalarMultiVector(RCP<RealValuedMultiVector> X);
                                                                                    ^
In file included from packages/muelu/src/MueLu_Utilities_kokkos.hpp:1:0,
                 from /home/mbetten/Trilinos/Trilinos/packages/muelu/src/Graph/MueLu_CoalesceDropFactory_kokkos_def.hpp:63,
                 from packages/muelu/src/Utils/ExplicitInstantiation/MueLu_CoalesceDropFactory_kokkos.cpp:55:
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Utils/MueLu_Utilities_kokkos_decl.hpp:415:63: error: ‘RealValuedMultiVector’ was not declared in this scope
     static RCP<MultiVector> RealValuedToScalarMultiVector(RCP<RealValuedMultiVector> X);
                                                               ^~~~~~~~~~~~~~~~~~~~~
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Utils/MueLu_Utilities_kokkos_decl.hpp:415:84: error: template argument 1 is invalid
     static RCP<MultiVector> RealValuedToScalarMultiVector(RCP<RealValuedMultiVector> X);
                                                                                    ^
In file included from /home/mbetten/Trilinos/Trilinos/packages/kokkos/core/src/Kokkos_Core_fwd.hpp:51:0,
                 from /home/mbetten/Trilinos/Trilinos/packages/kokkos/core/src/Kokkos_Core.hpp:50,
                 from /home/mbetten/Trilinos/Trilinos/packages/teuchos/kokkoscompat/src/KokkosCompat_ClassicNodeAPI_Wrapper.hpp:5,
                 from /home/mbetten/Trilinos/Trilinos/packages/tpetra/classic/NodeAPI/Kokkos_DefaultNode.hpp:47,
                 from /home/mbetten/Trilinos/Trilinos/packages/muelu/src/Headers/MueLu_ConfigDefs.hpp:54,
                 from packages/muelu/src/Utils/ExplicitInstantiation/MueLu_CoalesceDropFactory_kokkos.cpp:49:
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Graph/MueLu_CoalesceDropFactory_kokkos_def.hpp: In instantiation of ‘struct MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Build(MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level&) const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = int; DeviceType = Kokkos::OpenMP; typename DeviceType::memory_space = Kokkos::HostSpace; MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level = MueLu::Level]::<lambda(MueLu::CoalesceDropFactory_kokkos<double, int, int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP> >::LO)>’:
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Graph/MueLu_CoalesceDropFactory_kokkos_def.hpp:643:33:   required from ‘void MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Build(MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level&) const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = int; DeviceType = Kokkos::OpenMP; typename DeviceType::memory_space = Kokkos::HostSpace; MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level = MueLu::Level]’
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Utils/MueLu_ETI_4arg.hpp:42:3:   required from here
/home/mbetten/Trilinos/Trilinos/packages/kokkos/core/src/Kokkos_Macros.hpp:213:27: warning: ‘MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Build(MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level&) const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = int; DeviceType = Kokkos::OpenMP; typename DeviceType::memory_space = Kokkos::HostSpace; MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level = MueLu::Level]::<lambda(MueLu::CoalesceDropFactory_kokkos<double, int, int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP> >::LO)>’ has a field ‘MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Build(MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level&) const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = int; DeviceType = Kokkos::OpenMP; typename DeviceType::memory_space = Kokkos::HostSpace; MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level = MueLu::Level]::<lambda(MueLu::CoalesceDropFactory_kokkos<double, int, int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP> >::LO)>::<distFunctor capture>’ whose type uses the anonymous namespace [-Wsubobject-linkage]
   #define KOKKOS_LAMBDA [=]
                           ^
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Graph/MueLu_CoalesceDropFactory_kokkos_def.hpp:644:15: note: in expansion of macro ‘KOKKOS_LAMBDA’
               KOKKOS_LAMBDA(const LO row) {
               ^
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Graph/MueLu_CoalesceDropFactory_kokkos_def.hpp: In instantiation of ‘struct MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Build(MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level&) const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = long long int; DeviceType = Kokkos::OpenMP; typename DeviceType::memory_space = Kokkos::HostSpace; MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level = MueLu::Level]::<lambda(MueLu::CoalesceDropFactory_kokkos<double, int, long long int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP> >::LO)>’:
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Graph/MueLu_CoalesceDropFactory_kokkos_def.hpp:643:33:   required from ‘void MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Build(MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level&) const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = long long int; DeviceType = Kokkos::OpenMP; typename DeviceType::memory_space = Kokkos::HostSpace; MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level = MueLu::Level]’
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Utils/MueLu_ETI_4arg.hpp:42:3:   required from here
/home/mbetten/Trilinos/Trilinos/packages/kokkos/core/src/Kokkos_Macros.hpp:213:27: warning: ‘MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Build(MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level&) const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = long long int; DeviceType = Kokkos::OpenMP; typename DeviceType::memory_space = Kokkos::HostSpace; MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level = MueLu::Level]::<lambda(MueLu::CoalesceDropFactory_kokkos<double, int, long long int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP> >::LO)>’ has a field ‘MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Build(MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level&) const [with Scalar = double; LocalOrdinal = int; GlobalOrdinal = long long int; DeviceType = Kokkos::OpenMP; typename DeviceType::memory_space = Kokkos::HostSpace; MueLu::CoalesceDropFactory_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::Level = MueLu::Level]::<lambda(MueLu::CoalesceDropFactory_kokkos<double, int, long long int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP> >::LO)>::<distFunctor capture>’ whose type uses the anonymous namespace [-Wsubobject-linkage]
   #define KOKKOS_LAMBDA [=]
                           ^
/home/mbetten/Trilinos/Trilinos/packages/muelu/src/Graph/MueLu_CoalesceDropFactory_kokkos_def.hpp:644:15: note: in expansion of macro ‘KOKKOS_LAMBDA’
               KOKKOS_LAMBDA(const LO row) {
               ^
ninja: build stopped: subcommand failed.
dir-opt (gcc) 212 $ 


```


## Motivation and Context
<!---
How has this expectation failure affected you?  What are you trying to
accomplish?  Why do we need to address this?  What does it have to do with
anything?  Providing context helps us come up with a solution that is most
useful in the real world.
-->

## Definition of Done
<!---
Tell us what needs to happen.  If necessary, give us a task list along the
lines of:                                                   
- [ ] First do this.
- [ ] Then do that.
- [ ] Also this other thing.
-->

## Possible Solution
<!---
Not obligatory, but suggest a fix for the bug or documentation, or suggest
ideas on how to implement the addition or change.
-->

## Steps to Reproduce
<!---
Provide a link to a live example, or an unambiguous set of steps to reproduce
this issue.  Include code to reproduce, if relevant. 
1. Do this.
1. Do that.
1. Shake fist angrily at computer.
-->

## Your Environment
<!---
Include relevant details about your environment such that we can replicate this
issue.
-->
- **Relevant repo SHA1s:**  
- **Relevant configure flags or configure script:**  
- **Operating system and version:**  
- **Compiler and TPL versions:**  

## Related Issues
<!---
If applicable, let us know how this bug is related to any other open issues:
-->
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of 

## Additional Information
<!---
Anything else that might be helpful for us to know in addressing this issue:
* Configure log file:  
* Build log file:  
* Test log file:  
* When was the last time everything worked (date/time; SHA1s; etc.)?
* What did you do that made the bug rear its ugly head?
* Have you tried turning it off and on again?
-->

--- a/packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_kokkos_decl.hpp
+++ b/packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_kokkos_decl.hpp
@@ -53,6 +53,8 @@
 
 #include <KokkosCompat_ClassicNodeAPI_Wrapper.hpp>
 
+#include "Teuchos_ScalarTraits.hpp"
+
 #include "MueLu_Aggregates_kokkos_fwd.hpp"
 #include "MueLu_AmalgamationFactory_fwd.hpp" // FIXME_KOKKOS (once we have kokkos version)
 #include "MueLu_AmalgamationInfo_fwd.hpp"    // FIXME_KOKKOS (once we have kokkos version)
@@ -112,6 +114,8 @@ namespace MueLu {
     typedef typename DeviceType::execution_space                     execution_space;
     typedef Kokkos::RangePolicy<local_ordinal_type, execution_space> range_type;
     typedef Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType>      node_type;
+    typedef typename Teuchos::ScalarTraits<Scalar>::magnitudeType    real_type;
+    typedef Xpetra::MultiVector<real_type, LocalOrdinal, GlobalOrdinal, node_type> RealValuedMultiVector;
 
   private:
     // For compatibility

--- a/packages/muelu/src/Utils/MueLu_Utilities_kokkos_decl.hpp
+++ b/packages/muelu/src/Utils/MueLu_Utilities_kokkos_decl.hpp
@@ -110,6 +110,7 @@ namespace MueLu {
 
   public:
     typedef typename Teuchos::ScalarTraits<SC>::magnitudeType Magnitude;
+    typedef Xpetra::MultiVector<Magnitude,LO,GO,NO> RealValuedMultiVector;
 
 #ifdef HAVE_MUELU_EPETRA
     //! Helper utility to pull out the underlying Epetra objects from an Xpetra object
@@ -318,6 +319,7 @@ namespace MueLu {
     typedef int LocalOrdinal;
     typedef int GlobalOrdinal;
     typedef typename Teuchos::ScalarTraits<Scalar>::magnitudeType Magnitude;
+    typedef Xpetra::MultiVector<Magnitude,LocalOrdinal,GlobalOrdinal,Node> RealValuedMultiVector;
 
   private:
 #undef MUELU_UTILITIES_KOKKOS_SHORT

--- a/packages/muelu/src/Utils/MueLu_Utilities_kokkos_def.hpp
+++ b/packages/muelu/src/Utils/MueLu_Utilities_kokkos_def.hpp
@@ -600,7 +600,8 @@ namespace MueLu {
 
   template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
   RCP<Xpetra::MultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node> >
-  RealValuedToScalarMultiVector(RCP<Xpetra::MultiVector<double,LocalOrdinal,GlobalOrdinal,Node> > X) {
+  Utilities_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+  RealValuedToScalarMultiVector(RCP<RealValuedMultiVector > X) {
     RCP<Xpetra::MultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node> > Xscalar;
 #if defined(HAVE_XPETRA_TPETRA) && defined(HAVE_TPETRA_INST_COMPLEX_DOUBLE)
     using range_type = Kokkos::RangePolicy<LocalOrdinal, typename Node::execution_space>;
@@ -623,17 +624,10 @@ namespace MueLu {
     return Xscalar;
   }
 
-  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-  RCP<Xpetra::MultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node> >
-  Utilities_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
-  RealValuedToScalarMultiVector(RCP<Xpetra::MultiVector<double,LocalOrdinal,GlobalOrdinal,Node> > X) {
-    return MueLu::RealValuedToScalarMultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>(X);
-  }
-
   template <class Node>
   RCP<Xpetra::MultiVector<double,int,int,Node> >
   Utilities_kokkos<double,int,int,Node>::
-  RealValuedToScalarMultiVector(RCP<Xpetra::MultiVector<double,int,int,Node> > X) {
+  RealValuedToScalarMultiVector(RCP<Xpetra::MultiVector<Magnitude,int,int,Node> > X) {
     return X;
   }
 

